pkg/daemon/update.go

```diff
@@ -93,9 +99,57 @@ func getNodeRef(node *corev1.Node) *corev1.ObjectReference {
 	}
 }
 
-// finalizeAndReboot is the last step in an update(), and it can also
-// be called as a special case for the "bootstrap pivot".
-func (dn *Daemon) finalizeAndReboot(newConfig *mcfgv1.MachineConfig) (retErr error) {
+func reloadService(name string) error {
+	_, err := runGetOut("systemctl", "reload", name)
+	return err
+}
+
+// performPostConfigChangeAction takes action based on what postConfigChangeAction has been asked.
+// For non-reboot action, it applies configuration, updates node's config and state.
+// In the end uncordon node to schedule workload.
+// If at any point an error occurs, we reboot the node so that node has correct configuration.
+func (dn *Daemon) performPostConfigChangeAction(postConfigChangeActions []string, configName string) error {
+	if ctrlcommon.InSlice(postConfigChangeActionReboot, postConfigChangeActions) {
+		dn.logSystem("Rebooting node")
+		return dn.reboot(fmt.Sprintf("Node will reboot into config %s", configName))
+	}
+	if ctrlcommon.InSlice(postConfigChangeActionNone, postConfigChangeActions) {
+		dn.logSystem("Node has Desired Config %s, skipping reboot", configName)
+	}
+	if ctrlcommon.InSlice(postConfigChangeActionReloadCrio, postConfigChangeActions) {
+		serviceName := "crio"
+		if err := reloadService(serviceName); err != nil {
+			dn.logSystem("Reloading %s configuration failed, rebooting: %v", serviceName, err)
```

Comment:
feel like this would make more sense as:
`"Reloading %s configuration failed with error: %v"

Reply:
Hmm I'd like to highlight the fact that we need to reboot, how about something like: `Reloading %s configuration failed, node will reboot instead. Error: %v` ?

Reply:
that works too :smile: